sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicLogicalOperators.scala

```diff
@@ -352,6 +351,36 @@ case class Join(
   }
 }
 
+/**
+ * Append data to an existing table.
+ */
+case class AppendData(
+    table: NamedRelation,
+    query: LogicalPlan,
+    isByName: Boolean) extends LogicalPlan {
+  override def children: Seq[LogicalPlan] = Seq(query)
+  override def output: Seq[Attribute] = Seq.empty
+
+  override lazy val resolved: Boolean = {
+    query.output.size == table.output.size && query.output.zip(table.output).forall {
+      case (inAttr, outAttr) =>
+          inAttr.name == outAttr.name &&                // names must match
+          outAttr.dataType.sameType(inAttr.dataType) && // types must match
```

Comment:
we should use `DataType.equalsIgnoreCompatibleNullability`, otherwise the nullability check blew does not consider nullability of inner fields, like struct field, array element, etc.

Reply:
Good catch.

Reply:
Good catch, I agree.